runtime/BUILD.gn

```diff
@@ -87,6 +87,16 @@ source_set("runtime") {
     "//third_party/tonic",
   ]
 
+  # naifu: On Windows and Android (in debug mode) the engine finds the Dart snapshot
+  # data through symbols that are statically linked into the executable.
+  # On other platforms this data is obtained by a dynamic symbol lookup.
```

Comment:
> This comment doesn't really answer my question of why Windows is different than every other platform, it just asserts that it's true. Why is it the case that Windows needs a different mechanism for this than all other platforms? Why can't it do the same dynamic symbol lookup?

  

  # naifu: On Windows the engine finds the Dart snapshot data through symbols
  # that are statically linked into the executable. On other platforms this
  # data is obtained by a dynamic symbol lookup.
  # The current Windows platform uses static linking instead of dynamic lookups
  # because:
  #  1. The size of the executable on Windows is not as concerned as on mobile
  #     platforms.
  #  2. If it is a dynamic lookup method, you need to modify the properties of
  #     the memory page after the *.bin is dynamically loaded at runtime and
  #     mark it as executable. This is feasible (similar to V8 on Windows), but
  #     this increases the complexity of the implementation(These operations
  #     involve the use of some system file/memory privilege APIs and there is
  #     a risk of failure).
  #  3. In the executable can enjoy the advantages of the windows preload
  #     mechanism, speed up I/O.

Reply:
Moreover, in the latest code of Chromium, the default implementation of the release mode of v8 snapshot has been changed from .bin dynamic loading to static link into executable.